File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    let answer = 0;
    for(let i = 0; i < s.length; i++){
        const arr = s.split('');
        for(let j = 0; j < i; j++){
            arr.push(arr.shift());
        }
        const stack = [];
        let isCorect = true;
        for(item of arr){
            if(item === '[' || item === '{' || item === '('){
                stack.push(item);
            } else {
                if(stack.length === 0) {
                    isCorect = false;
                    break;
                }
                const last = stack[stack.length - 1];
                if((item === ']' && last === '[') || (item === '}' && last === '{') || (item === ')' && last === '(')){
                    stack.pop();
                } else {
                    isCorect = false;
                    break;
                }
            }
        }
        if(isCorect && stack.length === 0) answer += 1;
    }
    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    count = 0
    for x in range(len(s)):
        stack = []
        is_valid = True
        for i in range(x, x + len(s)):
            idx = i % len(s)
            c = s[idx]
            if c == '[' or c == '{' or c == '(':
                stack.append(c)
            else:
                if not stack:
                    is_valid = False
                    break
                elif c == ']' and stack[-1] == '[':
                    stack.pop()
                elif c == '}' and stack[-1] == '{':
                    stack.pop()
                elif c == ')' and stack[-1] == '(':
                    stack.pop()
                else:
                    is_valid = False
                    break
        if is_valid and not stack:
            count += 1
    return count